exchanges/asset/asset.go

```diff
@@ -20,37 +20,35 @@ type Item uint32
 // Items stores a list of assets types
 type Items []Item
 
-// Const vars for asset package
+// Supported Assets
 const (
-	Empty Item = 0
-	Spot  Item = 1 << iota
+	Empty Item = iota
+	Spot
 	Margin
 	CrossMargin
 	MarginFunding
 	Index
 	Binary
+	Futures // All Futures must come after this
 	PerpetualContract
 	PerpetualSwap
-	Futures
 	DeliveryFutures
 	UpsideProfitContract
 	DownsideProfitContract
 	CoinMarginedFutures
 	USDTMarginedFutures
 	USDCMarginedFutures
-	Options
-	OptionCombo
 	FutureCombo
-	LinearContract // Added to represent a USDT and USDC based linear derivatives(futures/perpetual) assets in Bybit V5
-	All
-
-	optionsFlag   = OptionCombo | Options
-	futuresFlag   = PerpetualContract | PerpetualSwap | Futures | DeliveryFutures | UpsideProfitContract | DownsideProfitContract | CoinMarginedFutures | USDTMarginedFutures | USDCMarginedFutures | LinearContract | FutureCombo
-	supportedFlag = Spot | Margin | CrossMargin | MarginFunding | Index | Binary | PerpetualContract | PerpetualSwap | Futures | DeliveryFutures | UpsideProfitContract | DownsideProfitContract | CoinMarginedFutures | USDTMarginedFutures | USDCMarginedFutures | Options | LinearContract | OptionCombo | FutureCombo
+	LinearContract // Derivatives with a linear Base (e.g. USDT or USDC)
+	Options        // All Options must come after this
+	OptionCombo
+	All // Must come immediately after all valid assets
```

Comment:
Happy to move comment to it's own line, but the original wording seems saner to me.
"always be placed" is strange because ... we're not always placing it anywhere.

Happy to consider anything if `come immediately after all valid assets` is really unclear.

Note: `All` is not `IsValid`, and that's what this is about. Added to comment.

Fixed https://github.com/gbjk/gocryptotrader/commit/5809bb525

Reply:
Yeah what you have done now is good 🌈